lib/Login.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:flutter/material.dart';
+import 'dart:convert';
+import 'app_config.dart';
+import 'main_common.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'Profile.dart';
+
+String name;
+String email;
+String imageUrl;
+
+GoogleSignIn _googleSignIn = GoogleSignIn(
+  scopes: [
+    'email',
+    'https://www.googleapis.com/auth/userinfo.profile',
+  ],
+);
+
+_handleSignIn() async {
+  try {
+    await _googleSignIn.signIn();
+  } catch (error) {
+    print(error);
+  }
+}
+
+Future<String> tokenFromAccount(GoogleSignInAccount account) async {
+  GoogleSignInAuthentication auth;
+  try {
+    auth = await account.authentication;
+    print('okay');
+  } catch (error) {
+    print('error');
+  }
+  return auth.idToken;
+}
+
+class Login extends StatefulWidget {
+  @override
+  _LoginState createState() => _LoginState();
+}
+
+class _LoginState extends State<Login> {
+  GoogleSignInAccount currentUser;
+  String id;
+
+  setCurrentUser(GoogleSignInAccount account) {
+    setState(() {
+      currentUser = account;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    currentUser = null;
+    id = null;
+    try {
+      _googleSignIn.signInSilently();
+    } catch (error) {
+      _googleSignIn.signIn();
+    }
+    _googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount account) {
+      setCurrentUser(account);
+      tokenFromAccount(currentUser).then((token) async {
+        return await authenticationRequest(
+            AppConfig.of(context).baseUrl, token);
+      }).then((response) {
+        var json = jsonDecode(response);
+        setState(() {
+          id = (json['id'] != null) ? 'id' : 'error';
```

Comment:
Here, you shouldn't make id equal to the string 'id' or error'. It should be equal to json['id'] : null